BlueprintUI/Tests/Element+Testing.swift

```diff
@@ -9,7 +9,11 @@ extension Element {
     ///
     /// - Returns: A layout result
     func layout(frame: CGRect, environment: Environment = .empty) -> LayoutResultNode {
-        layout(layoutAttributes: LayoutAttributes(frame: frame), environment: environment)
+        layout(
+            frame: frame,
+            environment: environment,
+            layoutMode: RenderContext.current?.layoutMode ?? .default
```

Comment:
Seems like the first important usage of the singleton here.

I think this means that any raw call with get the default (legacy). I wonder if these tests would be more clearly rephrased to accept a layout mode as a parameter.

The difference is that existing tests may continue to be written as before, but their caffeinated analogs will require a different spelling, like:

```swift
RenderContext(layoutMode: .caffeinated).perform {
  element.layout(...)
}
```

No biggie and no need for change. I imagine this clarifies down the road.